catalog_tools/utils/simulate_distributions.py

```diff
@@ -1,20 +1,32 @@
 import numpy as np
+from scipy import stats
 
 
-def simulate_magnitudes(n: int, beta: float, mc: float) -> np.ndarray:
+def simulate_magnitudes(n: int, beta: float, mc: float,
+                        mag_max: float = None) -> np.ndarray:
     """ Generates a vector of n elements drawn from an exponential distribution
     exp(-beta*M)
 
     Args:
         n:      number of sample magnitudes
         beta:   scale factor of the exponential distribution
-        mc:     completeness magnitude
+        mc:     cut-off magnitude
+        mag_max: maximum magnitude. If it is not None, the exponential
+                 distribution is truncated at mag_max.
 
     Returns:
         mags:   vector of length n of magnitudes drawn from an exponential
         distribution
     """
-
-    mags = np.random.exponential(1 / beta, n) + mc
+    if mag_max:
```

Comment:
Is there a reference of how exactly the distribution is truncated?

Reply:
maybe this? Do you want it in the code? 
https://en.wikipedia.org/wiki/Inverse_transform_sampling